File: skip-the-dishes-query-event-handler/src/main/java/br/com/skip/the/dishes/query/event/handler/commons/EventStoreBasicAuth.java
package br.com.skip.the.dishes.query.event.handler.commons;

import feign.auth.BasicAuthRequestInterceptor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class EventStoreBasicAuth extends BasicAuthRequestInterceptor {

    public EventStoreBasicAuth(
            @Value("${event.store.user}") String user,
            @Value("${event.store.password}") String password) {
        super(user, password);
    }

}

File: skip-the-dishes-query-event-handler/src/main/kotlin/br/com/skip/the/dishes/query/event/handler/commons/EventStoreBasicAuth.kt
package br.com.skip.the.dishes.query.event.handler.commons

import feign.auth.BasicAuthRequestInterceptor
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class EventStoreBasicAuth(
        @Value("\${event.store.user}") user: String,
        @Value("\${event.store.password}") password: String
) : BasicAuthRequestInterceptor(user, password)
